python/ray/util/sgd/v2/callbacks/logging.py

```diff
@@ -87,3 +121,125 @@ def handle_result(self, results: Optional[List[Dict]]):
             f.seek(0)
             json.dump(
                 loaded_results + [results_to_log], f, cls=SafeFallbackEncoder)
+
+
+class SGDSingleFileSingleWorkerLoggingCallback(
+        SGDSingleFileLoggingCallback, metaclass=abc.ABCMeta):
+    """Abstract SGD logging callback class.
+
+    Allows only for single-worker logging.
+
+    Args:
+        logdir (Optional[str]): Path to directory where the results file
+            should be. If None, will be set by the Trainer.
+        filename (Optional[str]): Filename in logdir to save results to.
+        workers_to_log (int): Worker index to log. By default, will log the
+            worker with index 0.
+    """
+
+    # it's still workers_to_log, not worker, for easier switching
+    # this is defined again for different type hints
+    def __init__(self,
+                 logdir: Optional[str] = None,
+                 filename: Optional[str] = None,
+                 workers_to_log: int = 0) -> None:
+        return super().__init__(
+            logdir=logdir, filename=filename, workers_to_log=workers_to_log)
+
+    def _validate_workers_to_log(self, workers_to_log) -> int:
+        if isinstance(workers_to_log, Iterable):
+            workers_to_log = list(workers_to_log)
+            if len(workers_to_log) > 1:
+                raise ValueError(
+                    f"{self.__class__.__name__} only supports logging "
+                    "from a single worker.")
+            elif len(workers_to_log) < 1:
+                raise ValueError(
+                    "At least one worker must be specified in workers_to_log.")
+            workers_to_log = workers_to_log[0]
+        if not isinstance(workers_to_log, int):
+            raise TypeError("workers_to_log must be an integer.")
+        return workers_to_log
+
+
+class TBXLoggerCallback(SGDSingleFileSingleWorkerLoggingCallback):
+    """Logs SGD results in TensorboardX format.
+
+    Args:
+        logdir (Optional[str]): Path to directory where the results file
+            should be. If None, will be set by the Trainer.
+        filename (Optional[str]): Ignored.
+        workers_to_log (int): Worker index to log. By default, will log the
+            worker with index 0.
+    """
+
+    _default_filename: Union[str, Path] = None
+    VALID_SUMMARY_TYPES: Tuple[type] = (int, float, np.float32, np.float64,
+                                        np.int32, np.int64)
+    IGNORE_KEYS: Set[str] = {PID, TIMESTAMP, TIME_TOTAL_S, TRAINING_ITERATION}
+
+    def __init__(self,
+                 logdir: Optional[str] = None,
+                 filename: Optional[str] = None,
+                 workers_to_log: int = 0) -> None:
+        if filename:
```

Comment:
If we're going to ignore `filename` argument then do we need to even accept it as an arg?

Reply:
As above